feature/quicksettings/src/main/java/com/google/jetpackcamera/feature/quicksettings/QuickSettingsComponents.kt

```diff
@@ -0,0 +1,267 @@
+                    close = { isOpen = false })
+            }
+        }
+    }
+}
+
+/**
+ * The UI component for quick settings when it is expanded.
+ */
+@Composable
+private fun ExpandedQuickSettingsUi(
+    lensFace: CameraLensFace = CameraLensFace.FRONT,
+    flashMode: CameraFlashMode = CameraFlashMode.OFF,
+    aspectRatio: CameraAspectRatio = CameraAspectRatio.NINE_SIXTEEN,
+    timer: CameraTimer = CameraTimer.OFF,
+    onLensFaceClick: (lensFace: CameraLensFace) -> Unit,
+    onFlashModeClick: (flashMode: CameraFlashMode) -> Unit,
+    onAspectRatioClick: (aspectRatio: CameraAspectRatio) -> Unit,
+    onTimerClick: (timer: CameraTimer) -> Unit,
+    close: () -> Unit
+) {
+    var selectedUiModel by remember {
+        mutableStateOf<QuickSettingsUiModel?>(null)
+    }
+    val quickSettingsUiModels: List<QuickSettingsUiModel> = getQuickSettingsUiModelList(
+        lensFace = lensFace,
+        flashMode = flashMode,
+        aspectRatio = aspectRatio,
+        timer = timer,
+        onLensFaceClick = onLensFaceClick,
+        onFlashModeClick = onFlashModeClick,
+        onAspectRatioClick = onAspectRatioClick,
+        onTimerClick = onTimerClick
+    )
+    val initialNumOfColumns =
+        min(
+            quickSettingsUiModels.size,
+            ((LocalConfiguration.current.screenWidthDp.dp - (dimensionResource(
+                id = R.dimen.quick_settings_ui_horizontal_padding
+            ) * 2)) /
+                    (dimensionResource(id = R.dimen.quick_settings_ui_item_icon_size) +
+                            (dimensionResource(id = R.dimen.quick_settings_ui_item_padding) * 2))).toInt()
+        )
+    Column(
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(horizontal = dimensionResource(id = R.dimen.quick_settings_ui_horizontal_padding))
+    ) {
+        if (selectedUiModel == null) {
+            LazyVerticalGrid(
+                modifier = Modifier.fillMaxWidth(),
+                columns = GridCells.Fixed(count = initialNumOfColumns)
+            ) {
+                items(items = quickSettingsUiModels) { item ->
+                    QuickSettingUiItem(
+                        modifier = Modifier.clickable(onClick = {
+                            selectedUiModel = item
+                        }),
+                        drawableResId = item
+                            .drawableResIds[item.currentQuickSettingEnum.getOrdinal()],
+                        textRes = item
+                            .textResIds[item.currentQuickSettingEnum.getOrdinal()],
+                        descriptionRes = item
+                            .descriptionResIds[item.currentQuickSettingEnum.getOrdinal()],
+                        isHighLighted = false
+                    )
+
+                }
+            }
+        } else {
+            val expandedNumOfColumns =
+                min(
+                    selectedUiModel!!.descriptionResIds.size,
```

Comment:
You shouldn't have to use !! here, since selectedUiModel is guaranteed to be not null here

Reply:
Android Studio is not letting me do that here. The only way to get rid of !! is to make other val and assign selectedUiModel!! to it. That would just make the code more confusing.